### PR TITLE
[OSF-7976] Branded Preprints Phase 2 fixes

### PR DESCRIPTION
**Pull request by binoculars**
## Purpose

Updates for API fields needed for preprints

## Changes

Preprint Provider Serializer:
- Change preprint_provider serializer field domain_redirect_enabled to BooleanField
- Add domain and domain_redirect_enabled to filterable fields
Populate Preprint Providers Script:
- Change how domains are formatted
- domain_redirect_enabled is now boolean and'ed with local.py setting to enable provider domains
Remove environment variable in docker-compose override (now unneeded)

## Ticket

https://openscience.atlassian.net/browse/OSF-7976

### CODE REVIEW

File: api/preprint_providers/serializers.py [hunk 1]
```diff
@@ -20,7 +22,7 @@ class PreprintProviderSerializer(JSONAPISerializer):
     email_support = ser.CharField(required=False, allow_null=True)
     example = ser.CharField(required=False, allow_null=True)
     domain = ser.CharField(required=False, allow_null=False)
-    domain_redirect_enabled = ser.CharField(required=False, allow_null=False)
+    domain_redirect_enabled = ser.BooleanField(required=True)
```
Comment (cwisecarver) [inline api/preprint_providers/serializers.py]: lolwut

Comment (cwisecarver) [inline api/preprint_providers/serializers.py]: :+1: making a boolean a boolean.

File: scripts/populate_preprint_providers.py [hunk 2]
```diff
@@ -72,8 +74,10 @@ def update_or_create(provider_data):
 
 
 def format_domain_url(domain):
-    return ''.join((PREPRINT_PROVIDER_DOMAINS['prefix'], str(domain), PREPRINT_PROVIDER_DOMAINS['suffix'])) if \
-        PREPRINT_PROVIDER_DOMAINS['enabled'] else ''
+    prefix = PREPRINT_PROVIDER_DOMAINS['prefix'] or PROTOCOL
```
Comment (cwisecarver) [inline scripts/populate_preprint_providers.py]: There might be a safer way to do this with `furl` or `urlparse` but this seems safe enough for now assuming we're fairly confident in our data.

File: docker-compose.override.yml [hunk 3]
```diff
@@ -34,8 +34,6 @@ services:
 #  preprints:
 #    volumes_from:
 #      - container:preprints-sync
-##     environment:
-##       - ENABLE_PROVIDER_DOMAINS=true
```
Comment (jamescdavis) [inline docker-compose.override.yml]: Ah, I was wondering what this was for...

File: api/preprint_providers/serializers.py [hunk 4]
```diff
@@ -9,7 +9,9 @@ class PreprintProviderSerializer(JSONAPISerializer):
     filterable_fields = frozenset([
         'name',
         'description',
-        'id'
+        'id',
+        'domain',
```
Comment (jamescdavis) [inline api/preprint_providers/serializers.py]: whoomp there it is

**Pull request merged**